addons/binding/org.openhab.binding.dlinkupnpcamera/src/main/java/org/openhab/binding/dlinkupnpcamera/handler/DlinkUpnpCameraHandler.java

```diff
@@ -0,0 +1,327 @@
+
+    private UpnpIOService service;
+    private ScheduledFuture<?> pollingJob;
+
+    private Runnable pollingRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                logger.debug("Polling...");
+
+                // First check if the camera is set in the UPnP service registry
+                // If not, set the thing state to OFFLINE and wait for the next poll
+                if (!isUpnpDeviceRegistered()) {
+                    logger.debug("UPnP device {} not yet registered", getUDN());
+                    updateStatus(ThingStatus.OFFLINE);
+                    return;
+                }
+
+                // Check if the camera can be joined
+                // If not, set the thing state to OFFLINE and do nothing else
+                updatePlayerState();
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    return;
+                }
+            } catch (Exception e) {
+                logger.debug("Exception during poll : {}", e);
+            }
+        }
+    };
+
+    @Override
+    public void initialize() {
+
+        logger.debug("initializing handler for thing {}", getThing().getUID());
+
+        if (getUDN() != null) {
+            updateStatus(ThingStatus.ONLINE);
+            onUpdate();
+            hostname = getThing().getProperties().get(DlinkUpnpCameraConfiguration.IP);
+            super.initialize();
+            logger.debug("Camera initialized.");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            logger.warn("Cannot initalize the camera. UDN not set.");
+        }
+    }
+
+    private boolean isUpnpDeviceRegistered() {
+        return service.isRegistered(this);
+    }
+
+    public DlinkUpnpCameraHandler(Thing thing, UpnpIOService upnpIOService,
+            DiscoveryServiceRegistry discoveryServiceRegistry) {
+        super(thing);
+
+        logger.debug("Creating a DlinkUpnpCameraHandler for thing '{}'", getThing().getUID());
+        if (upnpIOService != null) {
+            this.service = upnpIOService;
+        }
+    }
+
+    @Override
+    public String getUDN() {
+        return getConfigAs(DlinkUpnpCameraConfiguration.class).udn;
+    }
+
+    private String getUsername() {
+        return getConfigAs(DlinkUpnpCameraConfiguration.class).username;
+    }
+
+    private String getPassword() {
+        return getConfigAs(DlinkUpnpCameraConfiguration.class).password;
+    }
+
+    private String getCommandRequest() {
+        return getConfigAs(DlinkUpnpCameraConfiguration.class).commandRequest;
+    }
+
+    private String getImageRequest() {
+        return getConfigAs(DlinkUpnpCameraConfiguration.class).imageRequest;
+    }
+
+    private void updatePlayerState() {
+        try {
+            refreshImage();
+            boolean isAlive = Ping.checkVitality(hostname, 0, PING_TIMEOUT);
```

Comment:
you just refreshed the image, i.e. either communicated successfully with the camera or not. The additional ping does not add any information. It's actually the opposite: with the ping you only know that there is SOME device responding on that IP, you don't know if it actually _is_ a camera. So I really think it would be better to make refreshImage return something which gives you the same info that you wanted to achieve with the ping.